tools/gz.cc

```diff
@@ -549,27 +569,54 @@ bool ModelCommand::RunImpl()
 }
 
 /////////////////////////////////////////////////
-bool ModelCommand::ProcessSpawn(sdf::SDFPtr _sdf,
+bool ModelCommand::ProcessSpawn(const std::string &_sdfString,
     const std::string &_name, const ignition::math::Pose3d &_pose,
     transport::NodePtr _node)
 {
-  sdf::ElementPtr modelElem = _sdf->Root()->GetElement("model");
+  transport::PublisherPtr pub = _node->Advertise<msgs::Factory>("~/factory");
+  pub->WaitForConnection();
+
+  // To set the name of the model, we have to parse the file, set the name and
+  // reserialize to SDFormat. However, loading it using libsdformat will result
+  // in upconversion, which we would like to avoid because the resulting
+  // document will lose its OriginalVersion attribute.
```

Comment:
maybe add one more bit: "So instead we parse and edit the model name with TinyXML"

Reply:
https://github.com/osrf/gazebo/pull/2734/commits/60d082bcdde5f998286b46c7957ca14a2988b1cb